diff --git a/docroot/profiles/panopoly/modules/contrib/wysiwyg/editors/ckeditor.inc b/docroot/profiles/panopoly/modules/contrib/wysiwyg/editors/ckeditor.inc
index 8924010..b5312bb 100644
--- a/docroot/profiles/panopoly/modules/contrib/wysiwyg/editors/ckeditor.inc
+++ b/docroot/profiles/panopoly/modules/contrib/wysiwyg/editors/ckeditor.inc
@@ -82,7 +82,7 @@ function wysiwyg_ckeditor_version($editor) {
       fclose($library);
       // Version numbers need to have three parts since 3.0.1.
       $version[1] = preg_replace('/^(\d+)\.(\d+)$/', '${1}.${2}.0', $version[1]);
-      return $version[1] . '.' . $version[2];
+      return $version[1] . ((isset($version[2])) ? '.' . $version[2] : '');
     }
     $max_lines--;
   }
